src/Microsoft.ApplicationInsights/TelemetryClientExtensions.cs

```diff
@@ -53,7 +54,33 @@ public static IOperationHolder<T> StartOperation<T>(this TelemetryClient telemet
 
             if (string.IsNullOrEmpty(operationTelemetry.Context.Operation.Id) && !string.IsNullOrEmpty(operationId))
             {
-                operationTelemetry.Context.Operation.Id = operationId;
+                var isActivityAvailable = ActivityExtensions.TryRun(() =>
+                {
+                    if (Activity.DefaultIdFormat == ActivityIdFormat.W3C)
+                    {
+                        if (W3CUtilities.IsCompatibleW3CTraceID(operationId))
+                        {
+                            // If the user provided operationid is W3C Compatible, use it.
+                            operationTelemetry.Context.Operation.Id = operationId;
+                        }
+                        else
+                        {
+                            // If user provided operationid is not W3C compatible, generate a new one instead.
+                            // and store supplied value inside customproperty.
+                            operationTelemetry.Context.Operation.Id = W3CUtilities.GenerateTraceId();
```

Comment:
```suggestion
                            operationTelemetry.Context.Operation.Id = ActivityTraceId.CreateRandom().ToHexString()
```

Reply:
Will modify the W3Utilities method to internally use ActivityTraceId.CreateRandom() if Activity is available. Else fallback to current implementation.

Reply:
please deprecate the old one and use new one here